Framework/Algorithms/src/ReflectometryReductionOneAuto2.cpp

```diff
@@ -976,35 +1000,79 @@ void ReflectometryReductionOneAuto2::applyPolarizationCorrection(
-  Workspace_sptr sumWS = transGroup->getItem(0)->clone();
-
-  /// For this step to appear in the history of the output workspaces I need
-  /// to set child to false and work with the ADS
-  auto plusAlg = createChildAlgorithm("Plus");
-  plusAlg->setChild(false);
-  plusAlg->initialize();
-
-  for (size_t item = 1; item < transGroup->size(); item++) {
-    plusAlg->setProperty("LHSWorkspace", sumWS);
-    plusAlg->setProperty("RHSWorkspace", transGroup->getItem(item));
-    plusAlg->setProperty("OutputWorkspace", transSum);
-    plusAlg->execute();
-    sumWS = AnalysisDataService::Instance().retrieve(transSum);
+MatrixWorkspace_sptr ReflectometryReductionOneAuto2::getFloodWorkspace() {
+  std::string const method = getProperty("FloodCorrection");
+  if (method == "Workspace" && !isDefault("FloodWorkspace")) {
+    return getProperty("FloodWorkspace");
+  } else if (method == "ParameterFile") {
+    if (!isDefault("FloodWorkspace")) {
+      g_log.warning() << "Flood correction is performed using data in the "
+                         "Parameter File. Value of FloodWorkspace property is "
+                         "ignored."
+                      << std::endl;
+    }
+    MatrixWorkspace_sptr inputWS = getProperty("InputWorkspace");
+    auto const instrument = inputWS->getInstrument();
+    auto const floodRunParam = instrument->getParameterAsString("Flood_Run");
+    if (floodRunParam.empty()) {
+      throw std::invalid_argument(
+          "Instrument parameter file doesn't have the Flood_Run parameter.");
+    }
+    boost::regex separator("\\s*,\\s*|\\s+");
+    auto const parts = Strings::StrParts(floodRunParam, separator);
+    if (!parts.empty()) {
+      std::string fileName = floodRunParam;
+      try {
+        // If the first part is a number treat all parts as run numbers
+        boost::lexical_cast<size_t>(parts.front());
+        fileName = instrument->getName() + Strings::toString(parts);
+      } catch (boost::bad_lexical_cast) {
+        // Do nothing fileName == floodRunParam
+      }
+      auto alg = createChildAlgorithm("CreateFloodWorkspace");
+      alg->initialize();
+      alg->setProperty("Filename", fileName);
+      std::string const prefix("Flood_");
+      for (auto const prop :
+           {"StartSpectrum", "EndSpectrum", "ExcludeSpectra", "Background",
+            "CentralPixelSpectrum", "RangeLower", "RangeUpper"}) {
+        auto const param = instrument->getParameterAsString(prefix + prop);
+        if (!param.empty()) {
+          alg->setPropertyValue(prop, param);
+        }
+      }
+      alg->execute();
+      MatrixWorkspace_sptr out = alg->getProperty("OutputWorkspace");
+      return out;
+    }
   }
-  MatrixWorkspace_sptr result =
-      boost::dynamic_pointer_cast<MatrixWorkspace>(sumWS);
-  AnalysisDataService::Instance().remove(transSum);
-  return result;
+  return MatrixWorkspace_sptr();
+}
+
+void ReflectometryReductionOneAuto2::applyFloodCorrection(
```

Comment:
Doxugen missing here?

Reply:
`/// Apply flood correction`?

Reply:
The thing here is just that the two methods `applyFloodCorrection` and `applyFloodCorrections` are called very similar and it is easy to miss the difference then. They are more different than I would guess (which is perform a correction for a single FloodWorkspace and perform a correction for multiple FloodWorkspaces - not the goal).
But this is up to you to decide. Only my opinion.